tests/unit/test_response.py

```diff
@@ -72,20 +71,24 @@ def test_raw_response__reset(mock_session):
     assert response.raw.read(None) == b'mock response'
 
 
-def test_raw_response__decode(mock_session):
```

Comment:
Since these make real HTTP requests, could you move these tests to `tests/integration/test_cache.py`?

Reply:
More details, if you're interested:

The unit tests need to be able to pass without an internet connection or any services running. [`mock_session`](https://github.com/reclosedev/requests-cache/blob/3878e05ab0e6c93589b9d0a0ccd88fc8cda06c15/tests/conftest.py#L60-L84) is a fixture that uses mock requests, so that one is used for the unit tests.

[Pytest fixtures](https://docs.pytest.org/en/stable/fixture.html), by the way, are kind of like `setUp()` and `tearDown()` methods with the stdlib unittest, but with a lot more features.

The integration tests call real services running in Docker containers (httpbin, Redis, etc.), and the [`tempfile_session`](https://github.com/reclosedev/requests-cache/blob/3878e05ab0e6c93589b9d0a0ccd88fc8cda06c15/tests/conftest.py#L87-L98) fixture is used for those.

There's a bit of info about that in the [contributing guide](https://github.com/reclosedev/requests-cache/blob/master/CONTRIBUTING.md#testing), but I could probably document that better.

Let me know if you have any other questions or suggestions about the tests in general!

Reply:
I see, that makes sense. Thanks!

The contributing guide is already very useful in its current form, it's all documented pretty well. I'm just not used to strictly separating unit tests and integration tests, largely because I've been mostly writing tests for projects that didn't really depend on any external services so far.